app/sources/HFTemplateController.m

```diff
@@ -380,6 +380,26 @@ - (BOOL)readMacDate:(NSDate **)value forLabel:(NSString *)label {
     return YES;
 }
 
+- (NSString *)readFatDateWithLabel:(NSString *)label error:(NSString **)error {
+    int16_t val;
+    if (![self readInt16:&val forLabel:nil]) {
+        if (error) {
+            *error = @"Failed to read int16 bytes";
+        }
+        return nil;
+    }
+
+    int day = val & 0x1F;
+    int month = (val >> 5) & 0xF;
+    int year = 1980 + ((val >> 9) & 0x7F);
+    NSString *date = [NSString stringWithFormat:@"%d-%02d-%02d", year, month, day];
```

Comment:
The other date commands use NSDateFormatter so dates are properly localized. Should we do the same here? Since these dates only have day, month, and year, we can use NSDateComponents along with NSCalendar to create an NSDate. If you agree, I can do this in a followup change.

Reply:
The FAT spec indicates there is a time format to. Is the date only useful on its own? Are you using this for reading a FAT filesystem or values elsewhere?

Reply:
Regarding localization, the best is probably to be consistent with the other commands. Although I prefer the ISO 8601 format (YYYY-MM-DD). I'll let you make the changes.

I need the time format too. I will do it after.
Dates can be used alone.

Reply:
You're right, current format looks best and doesn't need localizing.